libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -339,14 +339,19 @@ void RangeFinder::detect_instance(uint8_t instance, uint8_t& serial_instance)
             }
         }
         break;
-    case Type::MBI2C:
+    case Type::MBI2C: {
+        uint8_t addr = AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR;
+        if (params[instance].address != 0) {
+            addr = params[instance].address;
+        }
         FOREACH_I2C(i) {
             if (_add_backend(AP_RangeFinder_MaxsonarI2CXL::detect(state[instance], params[instance],
-                                                                  hal.i2c_mgr->get_device(i, AP_RANGE_FINDER_MAXSONARI2CXL_DEFAULT_ADDR)))) {
+                                                                  hal.i2c_mgr->get_device(i, addr)))) {
```

Comment:
I2C addresses are defined from 0 to 127.
I set the initial value of the I2C address to -1.
I have done PRs in the past that if -1 is set, use the default I2C address.

Reply:
Sure.  Sadly in this case people's existing configurations will have `0` in them.  Given those might be in permanent storage, it's probably worth losing `0` and not breaking people's setups.